src/daft-sql/src/modules/aggs.rs

```diff
@@ -34,20 +34,50 @@ impl SQLModule for SQLModuleAggs {
 
 impl SQLFunction for AggExpr {
     fn to_expr(&self, inputs: &[FunctionArg], planner: &SQLPlanner) -> SQLPlannerResult<ExprRef> {
-        let inputs = self.args_to_expr_unnamed(inputs, planner)?;
-        to_expr(self, inputs.as_slice())
+        // COUNT(*) needs a bit of extra handling, so we process that outside of `to_expr`
+        if let AggExpr::Count(_, _) = self {
+            handle_count(inputs, planner)
+        } else {
+            let inputs = self.args_to_expr_unnamed(inputs, planner)?;
+            to_expr(self, inputs.as_slice())
+        }
     }
 }
 
+fn handle_count(inputs: &[FunctionArg], planner: &SQLPlanner) -> SQLPlannerResult<ExprRef> {
+    Ok(match inputs {
+        [FunctionArg::Unnamed(FunctionArgExpr::Wildcard)] => match planner.relation_opt() {
+            Some(rel) => {
+                let schema = rel.schema();
+                col(schema.fields[0].name.clone())
+                    .count(daft_core::count_mode::CountMode::All)
+                    .alias("count")
+            }
```

Comment:
This is ok for now, but we will need to change this behavior to enable "correct" counting behavior, when we do it for the non-SQL path as well (which is to add a special CountRows logical plan op instead of relying on our aggregation machinery to count the first column).